janusgraph-test/src/test/java/org/janusgraph/graphdb/idmanagement/VertexIDAssignerTest.java

```diff
@@ -31,58 +31,66 @@
 
 import java.time.Duration;
 import java.util.ArrayList;
 import java.util.Arrays;
-import java.util.Collection;
 import java.util.List;
+import java.util.stream.Stream;
 
 /**
  * @author Matthias Broecheler (me@matthiasb.com)
  */
-
-@RunWith(Parameterized.class)
 public class VertexIDAssignerTest {
 
-    final VertexIDAssigner idAssigner;
+    private VertexIDAssigner idAssigner;
 
-    @Parameterized.Parameters
-    public static Collection<Object[]> configs() {
-        final List<Object[]> configurations = new ArrayList<>();
+    public static Stream<Arguments> configs() {
+        final List<Arguments> configurations = new ArrayList<>();
 
         for (int maxPerPartition : new int[]{Integer.MAX_VALUE, 100, 300}) {
             for (int numPartitions : new int[]{2, 4, 10}) {
                 for (int[] local : new int[][]{null, {0,2, numPartitions}, {235,234,8}, {1,1,2}, {0,1<<(numPartitions-1),numPartitions}}) {
-                    configurations.add(new Object[]{numPartitions, maxPerPartition, local});
+                    configurations.add(Arguments.arguments(numPartitions, maxPerPartition, local));
                 }
             }
         }
 
-        return configurations;
+        return configurations.stream();
     }
 
-    private final long maxIDAssignments;
+    private long maxIDAssignments;
 
-    private final long numPartitionsBits;
+    private long numPartitionsBits;
 
     private enum CustomIdStrategy {
         LOW,
         HIGH
     }
 
+    private JanusGraph getInMemoryGraph(boolean allowSettingVertexId, boolean idsFlush) {
+        ModifiableConfiguration config = GraphDatabaseConfiguration.buildGraphConfiguration();
+        config.set(GraphDatabaseConfiguration.STORAGE_BACKEND, InMemoryStoreManager.class.getCanonicalName());
+        config.set(GraphDatabaseConfiguration.IDS_FLUSH, idsFlush);
+        config.set(GraphDatabaseConfiguration.IDAUTHORITY_WAIT, Duration.ofMillis(1L));
+        config.set(GraphDatabaseConfiguration.CLUSTER_MAX_PARTITIONS, 1<<numPartitionsBits);
+        config.set(GraphDatabaseConfiguration.ALLOW_SETTING_VERTEX_ID, allowSettingVertexId);
+        return JanusGraphFactory.open(config);
+    }
+
     /**
      *
      * @param numPartitionsBits The number of partitions bits to use. This means there are exactly (1<<numPartitionBits) partitions.
      * @param partitionMax The maximum number of ids that can be allocated per partition. This is artificially constrained by the MockIDAuthority
      * @param localPartitionDef This array contains three integers: 1+2) lower and upper bounds for the local partition range, and
      *                          3) the bit width of the local bounds. The bounds will be bit-shifted forward to consume the bit width
      */
-    public VertexIDAssignerTest(int numPartitionsBits, int partitionMax, int[] localPartitionDef) {
+    public void setup(int numPartitionsBits, int partitionMax, int[] localPartitionDef){
```

Comment:
I have no idea to clean up this class.

Reply:
You code changes the flow because this part of code if executed only once for all tests and you execute this part of code for each individual test. If the solution for [this comment](https://github.com/JanusGraph/janusgraph/pull/1327#discussion_r) works then you can apply the fix here also.